app/app.rb

```diff
@@ -0,0 +1,83 @@
+RACK_ENV ||= 'development'
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative 'data_mapper_setup'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+  set :session_secret, 'super secret'
+  use Rack::MethodOverride
+
+
+  get '/' do
+    'Hello Chitter!'
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :'users/new'
+  end
+
+  post '/users' do
+    @user = User.new(name: params[:name], username: params[:username], email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect('users/welcome')
+    else
+      flash.now[:notice] = "Password and confirmation password do not match"
+      erb :'users/new'
+    end
+  end
+
+  get '/users/welcome' do #this is eventually going to be the homescreen ('/peeps/list') where peeps are read from
```

Comment:
Metrics/LineLength: Line is too long. [114/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)